src/test/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilderNamespaceTest.java

```diff
@@ -0,0 +1,157 @@
+    FilePath manifestFile = setupManifest("random", "default");
+    String commandNameSpace = KubernetesEngineBuilder.addNamespace(manifestFile, "default");
+    assertEquals("default", commandNameSpace);
+  }
+
+  @Test
+  public void testAddBlankNamespaceWithMultipleManifestNamespaces()
+      throws IOException, InterruptedException {
+    FilePath manifestPath = Mockito.mock(FilePath.class);
+    FilePath testManifest = setupManifest("test", "test");
+    FilePath defaultManifest = setupManifest("default", "default");
+    Mockito.when(manifestPath.isDirectory()).thenReturn(true);
+    FilePath[] manifests = {testManifest, defaultManifest};
+    Mockito.when(manifestPath.list(anyString())).thenReturn(manifests);
+    String commandNamespace = KubernetesEngineBuilder.addNamespace(manifestPath, "");
+    assertEquals("", commandNamespace);
+  }
+
+  @Test
+  public void testAddCustomNamespaceWithMultipleManifestNamespaces()
+      throws IOException, InterruptedException {
+    FilePath manifestPath = Mockito.mock(FilePath.class);
+    FilePath testManifest = setupManifest("test", "custom");
+    FilePath defaultManifest = setupManifest("default", "custom");
+    Mockito.when(manifestPath.isDirectory()).thenReturn(true);
+    FilePath[] manifests = {testManifest, defaultManifest};
+    Mockito.when(manifestPath.list(anyString())).thenReturn(manifests);
+    String commandNamespace = KubernetesEngineBuilder.addNamespace(manifestPath, "custom");
+    assertEquals("custom", commandNamespace);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testDoCheckNamespaceNPEWithNull() {
+    DescriptorImpl descriptor = new DescriptorImpl();
+    descriptor.doCheckNamespace(null);
+  }
+
+  @Test
+  public void testDoCheckNamespaceOKWithEmptyString() {
+    DescriptorImpl descriptor = new DescriptorImpl();
+    FormValidation result = descriptor.doCheckNamespace("");
+    assertNotNull(result);
+    assertEquals(FormValidation.ok().getMessage(), result.getMessage());
+  }
+
+  @Test
+  public void testDoCheckNamespaceWithOKProperlyFormedString() {
+    DescriptorImpl descriptor = new DescriptorImpl();
+    FormValidation result = descriptor.doCheckNamespace("test-a-23-b");
+    assertNotNull(result);
+    assertEquals(FormValidation.ok().getMessage(), result.getMessage());
+  }
+
+  @Test
+  public void testDoCheckNamespaceErrorWithValidCharactersMalformedString() {
+    DescriptorImpl descriptor = new DescriptorImpl();
+    FormValidation result = descriptor.doCheckNamespace("-test");
+    assertNotNull(result);
+    assertEquals(Messages.KubernetesEngineBuilder_NamespaceInvalid(), result.getMessage());
+  }
+
+  @Test
+  public void testDoCheckNamespaceErrorWithInvalidCharacters() {
+    DescriptorImpl descriptor = new DescriptorImpl();
+    FormValidation result = descriptor.doCheckNamespace("*");
+    assertNotNull(result);
+    assertEquals(Messages.KubernetesEngineBuilder_NamespaceInvalid(), result.getMessage());
+  }
+
+  /**
```

Comment:
Description of method? Though I kind of inferred it from the params and return.

Reply:
Good catch, thanks